File: sdks/python/examples/lazai_inference_settlement.py
from alith import Agent, LazAIClient

# 1. Join the iDAO, register user wallet on LazAI and deposit fees (Only Once)
LAZAI_IDAO_ADDRESS = "0x34d9E02F9bB4E4C8836e38DF4320D4a79106F194"
client = LazAIClient()
# Add by the inference node admin
# client.add_inference_node(LAZAI_IDAO_ADDRESS, "http://127.0.0.1:8000", "pub key")
try:
    client.get_user(client.wallet.address)
except Exception:
    client.add_user(10000000)
    client.deposit_inference(LAZAI_IDAO_ADDRESS, 5000000)

# 2. Request the inference server with the settlement headers and DAT file id
file_id = 1
url = client.get_inference_node(LAZAI_IDAO_ADDRESS)[1]
agent = Agent(
    # Note: replace with your model here
    model="Qwen-2.5",
    # OpenAI-compatible inference server URL
    base_url=f"{url}/v1",
    # Extra headers for settlement and DAT file anchoring
    extra_headers=client.get_request_headers(LAZAI_IDAO_ADDRESS, file_id=file_id),
)
print(agent.prompt("What is Alith?"))

File: sdks/node/examples/lazai_inference_settlement.ts
import { ChainConfig, Client } from "alith/lazai";
import { Agent } from "alith";

const node = "0x34d9E02F9bB4E4C8836e38DF4320D4a79106F194";
const client = new Client(ChainConfig.local());
// Add by the inference node admin
// await client.addInferenceNode(node, "http://127.0.0.1:8000", "pub key")
await client.addUser(100_000_000);
await client.depositInference(node, 50_000_000);
console.log(
  "The inference account of user is",
  await client.getInferenceAccount(client.getWallet().address, node)
);
const fileId = 1;
const nodeInfo = await client.getInferenceNode(node);
const url = nodeInfo.url;
const agent = new Agent({
  // OpenAI-compatible inference server URL
  baseUrl: `${url}/v1`,
  // Extra headers for settlement and DAT file anchoring
  extraHeaders: await client.getRequestHeaders(node, BigInt(fileId)),
});
console.log(await agent.prompt("What is Alith?"));
